.github/actions/multi-approvers/src/multi-approvers.ts

```diff
@@ -0,0 +1,274 @@
+import { RestEndpointMethodTypes } from "@octokit/rest";
+import { RequestError } from "@octokit/request-error";
+import { Fail, Result, Success } from "./result";
+
+type PullRequestReview =
+  RestEndpointMethodTypes["pulls"]["listReviews"]["response"]["data"];
+type Octokit = ReturnType<typeof getOctokit>;
+export type EventName = "pull_request" | "pull_request_review";
+
+export function isEventName(v: string): v is EventName {
+  return ["pull_request", "pull_request_review"].includes(v);
+}
+
+const MIN_APPROVED_COUNT = 2;
+const APPROVED = "approved";
+const COMMENTED = "commented";
+const ALLOWED_TEAM_MEMBER_ROLES = ["maintainer", "member"];
+const ACTIVE = "active";
+
+interface MultiApproversParams {
+  eventName: EventName;
+  runId: number;
+  branch: string;
+  pullNumber: number;
+  repoName: string;
+  repoOwner: string;
+  token: string;
+  team: string;
+  octokitOptions?: OctokitOptions;
+  logDebug: (_: string) => void;
+  logInfo: (_: string) => void;
+}
+
+export class MultiApproversAction {
+  private readonly eventName: string;
+  private readonly runId: number;
+  private readonly branch: string;
+  private readonly pullNumber: number;
+  private readonly repoName: string;
+  private readonly repoOwner: string;
+  private readonly team: string;
+  private readonly octokit: Octokit;
+
+  constructor(params: MultiApproversParams) {
+    this.eventName = params.eventName;
+    this.runId = params.runId;
+    this.branch = params.branch;
+    this.pullNumber = params.pullNumber;
+    this.repoName = params.repoName;
+    this.repoOwner = params.repoOwner;
+    this.team = params.team;
+    this.logDebug = params.logDebug;
+    this.logInfo = params.logInfo;
+
+    this.octokit = getOctokit(params.token, params.octokitOptions);
+  }
+
+  // Set in the constructor.
+  private logDebug: (_: string) => void;
+
+  // Set in the constructor.
+  private logInfo: (_: string) => void;
+
+  // Tests whether the given login is an active member of the team.
+  private async isInternal(login: string): Promise<boolean> {
+    try {
+      const response = await this.octokit.rest.teams.getMembershipForUserInOrg({
+        org: this.repoOwner,
+        team_slug: this.team,
+        username: login,
+      });
+      return (
+        ALLOWED_TEAM_MEMBER_ROLES.includes(response.data.role) &&
+        response.data.state === ACTIVE
+      );
+    } catch (err) {
+      if (err instanceof RequestError && err.status === 404) {
+        this.logDebug(
+          `Received 404 testing membership; assuming user is not a member: ${JSON.stringify(
+            err,
+          )}`,
+        );
+        // We can get here for a few known reasons:
+        // 1) The user is not a member
+        // 2) The team does not exist
+        // 3) Invalid token
+        // In all these cases, it's safe to return false.
+        return false;
+      }
+      throw err;
+    }
+  }
+
+  // Returns the number of approvals from members in the given list.
```

Comment:
Please document more thoroughly in english words how this function actually computes the count.

Reply:
Done.